simulationworkflowschema/equation_of_state.py

```diff
@@ -150,6 +153,67 @@ class EquationOfState(ParallelSimulation):
     def normalize(self, archive, logger):
         super().normalize(archive, logger)
 
+        # TODO - Check beyond the failing old test for example from a parser
+
+        # find the input structure
+        if self.inputs:
+            flag_input_structure = False
+            for input_item in self.inputs:
+                # DEBUGGING: resolve input_item.section to its actual object
+                # in order to test non-proxy case with proxy data
+                # TODO - Remove this after proper testing is included
+                # logger.warning(f'Is proxy: {isinstance(input_item.section, MProxy)}') ## DEBUG
+                # input_item.section.m_resolved() ## DEBUG
+                # logger.warning(f'Is proxy: {isinstance(input_item.section, MProxy)}') ## DEBUG
+
+                # Old way to compare sections
+                input_proxy_value = ''
+                if isinstance(input_item.section, MProxy):
+                    input_proxy_value = input_item.section.m_proxy_value
+
+                section = (
```

Comment:
This approach is now compatible with non-proxy sections, I tested it by pre-resolving the sections (see the #DEBUGGING section code above)

Reply:
does `m_resolved()` not return also the section if not MProxy?

Reply:
idk, I will check that :saluting_face:  

Previously I was looking at this weird old test data which didn't have `m_resolved()` defined, so I think I got off track because of that